Library/Homebrew/formula_installer.rb

```diff
@@ -1324,7 +1367,7 @@ def pour
     end
 
     HOMEBREW_CELLAR.cd do
-      downloader.stage
+      T.cast(downloader, T.any(LocalBottleDownloadStrategy, Bottle)).stage
```

Comment:
This could technically also be a `Formula`, for which `stage` is a private method. I've used a `cast` here to minimize mucking around, but I'm open to other suggestions.

Reply:
https://github.com/Homebrew/brew/pull/17756 will probably help here